### PR TITLE
Implemented Ox parser for MathML parsing

### LINKED ISSUE #63: Plurimath parsing of MathML presupposes no namespacing
I'm putting workarounds, so this is not urgent, but:

Your code presupposes that any MathML it parses will not have namespacing, and namespace prefixes on nodes.

But of course, I will routinely want to grab a namespaced MathML node from a Metanorma document, and parse it in Plurimath. That means that what you will be seeing is:

`Plurimath::Math.parse("<m:math>\n  <m:msup> <m:mrow> <m:mo>(</m:mo> <m:mrow> <m:mi>x</m:mi> <m:mo>+</m:mo> <m:mi>y</m:mi> </m:mrow> <m:mo>)</m:mo> </m:mrow> <m:mn>2</m:mn> </m:msup>\n</m:math>"", "mathml")`

(where the `m:` prefix is defined in the document root, but doesn't necessarily make it into the XML string, depending on how I invoke it.)

And Plurimath can't cope with those node prefixes:

````
     Parslet::ParseFailed:
       Failed to match sequence ('<' (open:'annotation-xml' / open:'annotation_xml' / open:'mmultiscripts' / open:'maligngroup' / open:'malignmark' / open:'annotation' / open:'munderover' / open:'mscarries' / open:'semantics' / open:'mphantom' / open:'mlongdiv' / open:'menclose' / open:'mscarry' / open:'msubsup' / open:'mpadded' / open:'maction' / open:'msgroup' / open:'mfenced' / open:'merror' / open:'munder' / open:'mtable' / open:'mstyle' / open:'mstack' / open:'mspace' / open:'msline' / open:'mfrac' / open:'mover' / open:'msrow' / open:'mroot' / open:'msqrt' / open:'msup' / open:'msub' / open:'mrow' / open:'math' / open:'mtr' / open:'mtd' / open:'ms' / open:'mi' / open:'mo' / open:'mn') attributes:((name:([a-zA-Z]{0, }) '=' ('"' value:([^\"]{0, }) '"' / ''' value:([^']{0, }) ''')){0, }) '>' iteration:ITERATION '<' '/' (close:'annotation-xml' / close:'annotation_xml' / close:'mmultiscripts' / close:'maligngroup' / close:'malignmark' / close:'annotation' / close:'munderover' / close:'mscarries' / close:'semantics' / close:'mphantom' / close:'mlongdiv' / close:'menclose' / close:'mscarry' / close:'msubsup' / close:'mpadded' / close:'maction' / close:'msgroup' / close:'mfenced' / close:'merror' / close:'munder' / close:'mtable' / close:'mstyle' / close:'mstack' / close:'mspace' / close:'msline' / close:'mfrac' / close:'mover' / close:'msrow' / close:'mroot' / close:'msqrt' / close:'msup' / close:'msub' / close:'mrow' / close:'math' / close:'mtr' / close:'mtd' / close:'ms' / close:'mi' / close:'mo' / close:'mn') '>') at line 1 char 2.
````

I am stripping the `m:` prefixes by regex before calling Plurimath.parse. I shouldn't have to: either you strip them on receipt, or you allow optional prefixes on all element names.

### LINKED ISSUE #67: MathML: add in test examples from MathML 3.0 spec
Spec location: https://www.w3.org/TR/MathML3/

### LINKED ISSUE #75: (URGENT) Parse MathML using Ox
We should use native XML parsing via Ox for MathML.

### LINKED ISSUE #89: Conversion to Asciimath crashing on undefined map for fine space
From parsing the first document in bipm-si-brochure, https://github.com/metanorma/bipm-si-brochure/blob/main/sources/rec-values/mep-wgfs-nm237-115In_1267THz.adoc

Asciimath original: `f ~~ 1267 "unitsml(THz)"]`

````

math = "<math><mi>f</mi><mo>≈</mo><mn>1267</mn><mo rspace=\"thickmathspace\">⁢</mo><mrow xref=\"U_THz\"><mi mathvariant=\"normal\">THz</mi></mrow></math>"

Plurimath::Math.parse(math, "mathml").to_asciimath
````

which parses as

````
@value=
  [#<Plurimath::Math::Function::F:0x000000011e535498 @parameter_one=nil>,
   #<Plurimath::Math::Symbol:0x000000011e52d928 @value="~~">,
   #<Plurimath::Math::Symbol:0x000000011e526038 @value="1267">,
   {:mo=>{:attributes=>{:rspace=>"thickmathspace"}, :value=>["⁢"]}},
   #<Plurimath::Math::Formula:0x000000011e50eeb0
    @value=[#<Plurimath::Math::Symbol:0x000000011e514c20 @value="THz">],
    @wrapped=false>],
````

 crashes with:

````
eval error: undefined method `to_asciimath' for {:mo=>{:attributes=>{:rspace=>"thickmathspace"}, :value=>["⁢"]}}:Hash

          value.map(&:to_asciimath).join
               ^^^^
  /Users/nickn/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/plurimath-0.2.1/lib/plurimath/math/formula.rb:21:in `map'
  /Users/nickn/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/plurimath-0.2.1/lib/plurimath/math/formula.rb:21:in `to_asciimath'
````

It is imperative that all Plurimath conversions degrade gracefully; otherwise, they simply cannot be used. 

In this instance, for example, MathML has injected into the expression a fine space, and Asciimath may not have an exact equivalent; if not, it needs to fall back to a generic space.

### PR DESCRIPTION
**Pull request by suleman-uzair**
Using **Ox** parser for MathML parsing and new/updated **Parslet** transformation rules.

closes #63
closes #67  
closes #75 
closes #89

### CODE REVIEW

File: lib/plurimath/mathml/parser.rb [hunk 1]
```diff
@@ -9,17 +9,38 @@ class Parser
...
+        @text = text
       end
 
       def parse
-        tree_t = Plurimath::Mathml::Parse.new.parse(text)
-        tree_t = JSON.parse(tree_t.to_json, symbolize_names: true)
-        formula = Plurimath::Mathml::Transform.new.apply(tree_t)
-        formula = [formula] unless formula.is_a?(Array) || formula.nil?
-        return if formula.nil?
+        nodes = JSON.parse(
```
Comment (ronaldtse) [inline lib/plurimath/mathml/parser.rb:16]: This seems to be a performance issue, if you're converting Ox nodes into JSON, and then parsing JSON into a Hash. Can't we directly convert the Ox nodes into a Hash?

File: lib/plurimath/math/function/table.rb [hunk 2]
```diff
@@ -30,8 +30,8 @@ def to_mathml_without_math_tag
               column_lines << "none"
             end
           end
-          columnlines_attrs = "columnlines=\"#{column_lines.join(' ')}\"" if column_lines.include?("solid")
-          table_tag = "<mtable #{columnlines_attrs}>#{table_value}</mtable>"
+          columnlines_attrs = " columnlines=\"#{column_lines.join(' ')}\"" if column_lines.include?("solid")
+          table_tag = "<mtable#{columnlines_attrs}>#{table_value}</mtable>"
```
Comment (ronaldtse) [inline lib/plurimath/math/function/table.rb:34]: Will these go away by using Ox to manipulate nodes?

Comment (suleman-uzair) [reply lib/plurimath/math/function/table.rb:34]: Yes, these will go away in the reverse conversion using Ox Parser. I’m currently working on it.

**Review by ronaldtse — APPROVED** [review]
Thank you @suleman-uzair !

**Pull request merged**